ts/src/pro/blofin.ts

```diff
@@ -0,0 +1,489 @@
+
+//  ---------------------------------------------------------------------------
+
+import blofinRest from '../blofin.js';
+import { NotSupported, ArgumentsRequired } from '../base/errors.js';
+import { ArrayCache, ArrayCacheByTimestamp } from '../base/ws/Cache.js';
+import type { Int, MarketInterface, Trade, OrderBook, Strings, Ticker, Tickers, OHLCV } from '../base/types.js';
+import Client from '../base/ws/Client.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class blofin extends blofinRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchTrades': true,
+                'watchTradesForSymbols': true,
+                'watchOrderBook': true,
+                'watchOrderBookForSymbols': true,
+                'watchTicker': true,
+                'watchTickers': true,
+                'watchOHLCV': true,
+                'watchOHLCVForSymbols': true,
+            },
+            'urls': {
+                'api': {
+                    'ws': {
+                        'swap': {
+                            'public': 'wss://openapi.blofin.com/ws/public',
+                            'private': 'wss://openapi.blofin.com/ws/private',
```

Comment:
aren't you going to add the private part? 
![image](https://github.com/ccxt/ccxt/assets/43336371/f7f6c36c-625f-42fa-a9ee-5c8e31d11a0e)

Reply:
there is no private ws method supported by exchange, probably it's reserved for future

Reply:
u sure? they have private streams documented
![image](https://github.com/ccxt/ccxt/assets/43336371/519740a6-f96c-4b24-bcb1-0e26d4a5b082)

Reply:
ouch, i haven't had found that under tabs, now i saw, i missed the "trading" tab :/ 
hmm. may I do private part in separate PR after this?

Reply:
@ttodua no, you can add it here

Reply:
ah ok ✌️